rerun_py/rerun_sdk/rerun/components/view_coordinates_ext.py

```diff
@@ -0,0 +1,50 @@
+from __future__ import annotations
+
+from enum import IntEnum
+from typing import TYPE_CHECKING, Iterable
+
+import numpy as np
+import pyarrow as pa
+
+if TYPE_CHECKING:
+    from ..log import ComponentBatchLike
+    from . import ViewCoordinatesArrayLike
+
+
+class ViewCoordinatesExt:
+    class ViewDir(IntEnum):
+        Up = 1
+        Down = 2
+        Right = 3
+        Left = 4
+        Forward = 5
+        Back = 6
+
+    @staticmethod
+    def native_to_pa_array_override(data: ViewCoordinatesArrayLike, data_type: pa.DataType) -> pa.Array:
+        from . import ViewCoordinates
+
+        if isinstance(data, ViewCoordinates):
+            data = data.coordinates
+
+        data = np.asarray(data, dtype=np.uint8)
+
+        if data.shape != (3,):
+            raise ValueError(f"ViewCoordinates must be a 3-element array. Got: {data.shape}")
+
+        for value in data:
+            # TODO(jleibs): Enforce this validation based on ViewDir
+            if value not in range(1, 7):
+                raise ValueError("ViewCoordinates must contain only values in the range [1,6].")
+
+        return pa.FixedSizeListArray.from_arrays(data, type=data_type)
```

Comment:
Unless I'm misreading this, this feels wrong to me.
The component has no idea whether it should be mono or not: that's a property of how the archetype was defined.
If there happens to be multiple instances of ViewCoordinates, then this should serialize all of them without question.

What if something tries to define a custom archetype that actually uses a batch, or even just logs a batch directly using the low level APIs?

Reply:
Ah yeah, you're right. Good catch.